xdem/dem.py

```diff
@@ -1,76 +1,92 @@
-    # ASTGTM2/ASTGTM3: https://lpdaac.usgs.gov/documents/434/ASTGTM_User_Guide_V3.pdf
-    # NASADEM: https://lpdaac.usgs.gov/documents/592/NASADEM_User_Guide_V1.pdf, HGTS is ellipsoid, HGT is EGM96 geoid !!
-    # ArcticDEM (mosaic and strips): https://www.pgc.umn.edu/data/arcticdem/
-    # REMA (mosaic and strips): https://www.pgc.umn.edu/data/rema/
-    # TanDEM-X 90m global: https://geoservice.dlr.de/web/dataguide/tdm90/
-    # COPERNICUS DEM: https://spacedata.copernicus.eu/web/cscda/dataset-details?articleId=394198
-
-    if product in ["ArcticDEM/REMA", "TDM1", "NASADEM-HGTS"]:
-        vref_name = "WGS84"
-    elif product in ["AW3D30", "SRTMv4.1", "SRTMGL1", "ASTGTM2", "NASADEM-HGT"]:
-        vref_name = "EGM96"
-    elif product in ["COPDEM"]:
-        vref_name = "EGM08"
-    else:
-        vref_name = None
-
-    return vref_name
-
-
-dem_attrs = ["vref", "vref_grid", "_ccrs"]
-
 
-class DEM(SatelliteImage):  # type: ignore
     def __init__(
         self,
         filename_or_dataset: str | RasterType | rio.io.DatasetReader | rio.io.MemoryFile,
-        vref_name: str | None = None,
-        vref_grid: str | None = None,
+        vcrs: Literal["Ellipsoid"]
+        | Literal["EGM08"]
+        | Literal["EGM96"]
+        | VerticalCRS
+        | str
+        | pathlib.Path
+        | int
+        | None = None,
         silent: bool = True,
         **kwargs: Any,
     ) -> None:
         """
-        Load digital elevation model data through the Raster class, parse additional attributes from filename or
-        metadata through the SatelliteImage class, and then parse vertical reference from DEM product name.
-        For manual input, only one of "vref", "vref_grid" or "ccrs" is necessary to set the vertical reference.
+        Instantiate a digital elevation model.
+
+        The vertical reference of the DEM can be defined by passing the `vcrs` argument.
+        Otherwise, a vertical reference is tentatively parsed from the DEM product name.
+
+        Inherits all attributes from the :class:`geoutils.Raster` and :class:`geoutils.SatelliteImage` classes.
 
         :param filename_or_dataset: The filename of the dataset.
-        :param vref_name: Vertical reference name
-        :param vref_grid: Vertical reference grid (any grid file in https://github.com/OSGeo/PROJ-data)
-        :param silent: Whether to display vertical reference setting
-        :param silent: boolean
+        :param vcrs: Vertical coordinate reference system either as a name ("WGS84", "EGM08", "EGM96"),
+            a EPSG code or pyproj.crs.VerticalCRS, or a path to a PROJ grid file (https://github.com/OSGeo/PROJ-data).
+        :param silent: Whether to display vertical reference parsing.
         """
 
         self.data: NDArrayf
+        self._vcrs: VerticalCRS | None = None
+        self._vcrs_name: str | None = None
+        self._vcrs_grid: str | None = None
+        self._ccrs: CompoundCRS | None = None
```

Comment:
I can't find where `self._ccrs` is ever used. Doesn't the property always create a new one?

Reply:
Good point, removed!